### PR TITLE
refactor: `device` lib file

### LINKED ISSUE #5080: [Task]: Refactor `device.ts` lib file
### Task description

There is some existing older code in the root shared `lib` folder that needs to be refactored into modules.

### Requirements

- [x] Create new e.g. `scanner` auxiliary module in `packages/mobile`
- [x] Move the QR scanner code (`lib/device.ts`) to new module
- [x] Remove `BarcodeManager` from `IPlatform` interface

### Creation checklist

- [X] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### PR DESCRIPTION
**Pull request by maxwellmattryan**
## Summary

> Please summarize your changes, describing __what__ they are and __why__ they were made.

This code is old and should be cleaned up since it will be used later.

## Changelog

```
- Create auxiliary scanner module in mobile
- Remove old device lib file
```

## Relevant Issues

> Please list any related issues using [development keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

Closes #5080

### Platforms

> Please select any platforms where your changes have been tested.

- __Desktop__
  - [ ] MacOS
  - [x] Linux
  - [ ] Windows
- __Mobile__
  - [ ] iOS
  - [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to __test and verify__ that your changes work as intended.

Test general app functionality.

## Checklist

> Please tick all of the following boxes that are relevant to your changes.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/mobile/lib/auxiliary/scanner/index.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export * from './actions'
+export * from './interfaces'
+export * from './stores'
```
Comment (begonaalvarezd) [inline packages/mobile/lib/auxiliary/scanner/index.ts:3]: shouldnt we export also `helpers` here?

Comment (maxwellmattryan) [reply packages/mobile/lib/auxiliary/scanner/index.ts:3]: No, helpers are supposed to be only for internal use in the same module.

File: packages/mobile/components/QRScanner.svelte [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <script lang="typescript">
-    import { Icon } from 'shared/components'
-    import { showCameraScanner, stopQRScanner } from 'shared/lib/device'
+    import { Icon } from '@components'
+    import { showCameraScanner, stopQrScanner } from '@auxiliary/scanner'
```
Comment (begonaalvarezd) [inline packages/mobile/components/QRScanner.svelte:3]: atm this breaks, we cant use aliases in mobile 🙏🏼

Comment (begonaalvarezd) [reply packages/mobile/components/QRScanner.svelte:3]:
```suggestion
    import { showCameraScanner, stopQrScanner } from '../lib/auxiliary/scanner'
```

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]
Looking great but the code doesnt work anymore, so we need to tweak some things, for now I can see we could update this: 
- [x] `packages/mobile/capacitor/lib/barcodeManager.ts` we moved `IBarcodeManager` to `packages/mobile/lib/auxiliary/scanner/helpers/barcode-manager.helper.ts`

**Review by begonaalvarezd — APPROVED** [review]

**Pull request merged**